packages/dds/tree/src/test/feature-libraries/editable-tree/utils.ts

```diff
@@ -56,21 +59,31 @@ export function expectTreeEquals(
     assert.equal(node[valueSymbol], expected.value);
     const type = node[getTypeSymbol](undefined, false);
     assert.deepEqual(type, expectedType);
-    for (const key of Object.keys(node)) {
-        const subNode: UnwrappedEditableField = node[key as FieldKey];
-        assert(subNode !== undefined, key);
-        const fields = expected.fields ?? {};
-        assert.equal(key in fields, true);
-        const field: JsonableTree[] = fields[key];
+    for (const ok of Reflect.ownKeys(node)) {
+        const key: FieldKey = brand(ok);
```

Comment:
I think this comparison does not detect fields that are in the jsonableTree but missing from EditiableTrees. We can fix this by removing each field as we see it from an expectedField array:

```suggestion
    const expectedFields = new Set(genericTreeKeys(node));
    for (const ok of Reflect.ownKeys(node)) {
        const key: FieldKey = brand(ok);
        assert(expectedFields.delete(key);
```

then assert the set is empty after the loop ends.

Reply:
Thanks, have missed that. Fixed.